### PR TITLE
378 deploy script error

### PR DESCRIPTION
**Pull request by aliciaaevans**
- Fix https://github.com/lcdb/lcdb-wf/issues/378 by passing in the source dir
- `flavor `wasn't properly being passed in and compared, so all workflows were being deployed
- The `--staging` default value was causing it to be considered always set. Instead of using a default on the arg, only use the default if `--clone` is set.
- improve documentation and comments in `test/lcdb-wf-test` and `deploy.py`

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -107,11 +107,11 @@ def write_include_file(source, flavor=None):
...
 
     patterns = []
-    if flavor is None or 'rnaseq':
+    if flavor is None or flavor == 'rnaseq':
         patterns.extend(PATTERN_DICT['rnaseq'])
-    if flavor is None or 'chipseq':
+    if flavor is None or flavor == 'chipseq':
         patterns.extend(PATTERN_DICT['chipseq'])
-    if flavor is None or 'full':
+    if flavor is None or flavor == 'full':
```
Comment (aliciaaevans) [inline deploy.py:114]: @daler Should full include chipseq and rnaseq?

Comment (daler) [reply deploy.py]: ah, yep, just pushed an attempted fix

**Comment by daler** [comment]
@aliciaaevans OK to merge this?

**Comment by aliciaaevans** [comment]
> @aliciaaevans OK to merge this?

@daler yes, everything looks good to me.

**Pull request merged**